src/main/java/webserver/request/Query.java

```diff
@@ -0,0 +1,47 @@
+package webserver.request;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Query {
+
+    private final Map<String, String> queryMap;
+
+    public Query(Map<String, String> queryMap) {
+        this.queryMap = queryMap;
+    }
+
+    public static Query empty(){
+        return new Query(new HashMap<>());
+    }
+
+    public static Query from(String queryStr) {
+        Map<String, String> queryMap = Arrays.stream(queryStr.split("&"))
+                .map(querySplit -> {
+                    String[] queryKV = querySplit.split("=");
+                    return Map.entry(queryKV[0], queryKV[1]);
+                }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        return new Query(queryMap);
+    }
+
+    public Map<String, String> getQueryMap() {
+        return queryMap;
```

Comment:
```suggestion
        return new HashMap<String, String>(queryMap);
```
컬렉션에 대한 방어적 복사를 활용해보는 것도 좋을 것 같아요 :)